services/net/content/ContentManager.cs

```diff
@@ -149,8 +149,11 @@ private void ConsumeMessages()
     {
         if (_consumer == null || _notRunning.Contains(_consumer.Status))
         {
+            // Make sure the prior task is cancelled before creating a new one.
+            if (_cancelToken?.IsCancellationRequested == false)
+                _cancelToken?.Cancel();
             _cancelToken = new CancellationTokenSource();
-            _consumer = ConsumerHandlerAsync();
+            _consumer = Task.Run(ConsumerHandlerAsync, _cancelToken.Token);
```

Comment:
I originally switch away from `Task.Run(...)`, but I don't remember the reason.  `Factory.StartNew(...)` seemed to be a good choice, but it was also not awaiting a Task that performed async methods inside it.  So far my testing appears to show this working as intended.